AttributeKeyProvider.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+namespace Pim\Bundle\DataGeneratorBundle;
+
+use Pim\Component\Catalog\Model\AttributeInterface;
+use Pim\Component\Catalog\Model\ChannelInterface;
+use Pim\Component\Catalog\Model\CurrencyInterface;
+use Pim\Component\Catalog\Model\LocaleInterface;
+use Pim\Component\Catalog\Repository\ChannelRepositoryInterface;
+use Pim\Component\Catalog\Repository\CurrencyRepositoryInterface;
+use Pim\Component\Catalog\Repository\LocaleRepositoryInterface;
+
+/**
+ * @author    Philippe Mossière <philippe.mossiere@akeneo.com>
+ * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class AttributeKeyProvider
+{
+    const METRIC_UNIT = 'unit';
+
+    /** @var ChannelRepositoryInterface */
+    private $channelRepository;
+
+    /** @var LocaleRepositoryInterface */
+    private $localeRepository;
+
+    /** @var CurrencyRepositoryInterface */
+    private $currencyRepository;
+
+    /** @var ChannelInterface[] */
+    private $channels;
+
+    /** @var CurrencyInterface[] */
+    private $currencies;
+
+    /** @var LocaleInterface[] */
+    private $locales;
+
+    /**
+     * ProductValueRawBuilder constructor.
+     *
+     * @param ChannelRepositoryInterface  $channelRepository
+     * @param LocaleRepositoryInterface   $localeRepository
+     * @param CurrencyRepositoryInterface $currencyRepository
+     */
+    public function __construct(
+        ChannelRepositoryInterface $channelRepository,
+        LocaleRepositoryInterface $localeRepository,
+        CurrencyRepositoryInterface $currencyRepository
+    ) {
+        $this->channelRepository = $channelRepository;
+        $this->localeRepository = $localeRepository;
+        $this->currencyRepository = $currencyRepository;
+    }
+
+    /**
+     * Generate the list of attribute keys for a given attribute
+     *
+     * Example:
+     * Attribute: description localizable in english and french for ecommerce
+     * Attributes keys are:
+     * [ description-en_US-ecommerce, description-fr_FR-ecommerce ]
+     *
+     * @param AttributeInterface $attribute
+     *
+     * @return array
+     */
+    public function getAttributeKeys(AttributeInterface $attribute)
```

Comment:
quite unreadable IMHO. 70 lines, too much imbrication, no example for each case in description...